File: apexpathing/src/main/java/com/apexpathing/localization/FieldCoordinates.java
package com.apexpathing.localization;

import com.apexpathing.util.math.Pose;

/**
 * FTC field coordinate constants and utilities.
 */
public class FieldCoordinates {
    public static final double FIELD_SIZE = 144.0;
    public static final double HALF_FIELD = FIELD_SIZE / 2;

    public static final double MAX_X = HALF_FIELD;
    public static final double MIN_X = -HALF_FIELD;
    public static final double MAX_Y = HALF_FIELD;
    public static final double MIN_Y = -HALF_FIELD;

    public static Pose clamp(Pose pose) {
        return new Pose(
            Math.max(MIN_X, Math.min(MAX_X, pose.x())),
            Math.max(MIN_Y, Math.min(MAX_Y, pose.y())),
            pose.heading()
        );
    }
}

File: TeamCode/src/main/java/com/xpathing/main/localization/FieldCoordinates.kt
package com.xpathing.main.localization

import com.xpathing.main.follower.paths.Pose


object FieldCoordinates {
    const val fieldSize = 144.0
    const val halfField = fieldSize / 2

    // Field boundaries
    const val maxX = halfField
    const val minX = -halfField
    const val maxY = halfField
    const val minY = -halfField

    fun clamp(pose: Pose): Pose {
        return Pose(
            pose.x.coerceIn(minX, maxX),
            pose.y.coerceIn(minY, maxY),
            pose.heading
        )
    }

}